src/sonic_ax_impl/mibs/__init__.py

```diff
@@ -160,6 +160,30 @@ def mgmt_if_entry_table_state_db(if_name):
 
     return b'MGMT_PORT_TABLE|' + if_name
 
+def get_sai_id_key(namespace, sai_id):
+    """
+    inputs:
+    namespace - string
+    sai id - bytes
+    Return type:
+    bytes
+    Return value: namespace:sai id or sai id
+    """
+    if namespace != '':
+        return  namespace.encode() + b':' + sai_id
+    else:
+        return sai_id
+
+def split_sai_id_key(sai_id_key):
+    """
+    Input - bytes
+    Return namespace string and sai id in byte string.
+    """
+    if b':' in sai_id_key:
+        namespace, sai_id = sai_id_key.split(b':')
+        return namespace.decode(), sai_id
```

Comment:
you can split first and check result len().
This will scan the  string only once.

Reply:
Updated as per comment.